hasher.go

```diff
@@ -270,6 +270,11 @@ func (h *Hasher) Merkleize(indx int) {
 func (h *Hasher) MerkleizeWithMixin(indx int, num, limit uint64) {
 	input := h.buf[indx:]
 
+	if rest := len(input) % 32; rest != 0 {
+		// pad zero bytes to the left
+		input = append(input, zeroBytes[:32-rest]...)
+	}
```

Comment:
Related to comment: https://github.com/ferranbt/fastssz/pull/120/files#r1097352609 

`PutBytes` and `MerkleizeWithMixin` cannot be used together because they both merklize bytes, resulting in double merklization. The correct use is `Append` with `MerkleizeWithMixin`. However, `MerkleizeWithMixin` should then check if the bytes should be padded to the next 32 bytes increment.

Reply:
Could you use `FillUpTo32` instead at the beginning of the function for the whole `h.buf` array?

Reply:
Good point! Addressed in https://github.com/ferranbt/fastssz/pull/120/commits/b74a6c6194e0cef349a74eebcf4f8ab873c5955e.